src/app/modules/home/components/login-form/login-form.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, OnInit } from "@angular/core";
+import { Router, ActivatedRoute } from "@angular/router";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { first } from "rxjs/operators";
+import { AuthService } from "../../../../core/services/auth/auth.service";
+import { AlertService } from "../../../../core/services/alert/alert.service";
+import { MatTabChangeEvent } from "@angular/material/tabs";
+
+@Component({
+  selector: "app-login-form",
+  templateUrl: "./login-form.component.html",
+  styleUrls: ["./login-form.component.scss"]
+})
+export class LoginFormComponent implements OnInit {
+  registerForm: FormGroup;
+  loginForm: FormGroup;
+  loading = false;
+  submitted = false;
+  returnUrl: string;
+  selectedTabIndex: number;
+
+  constructor(
+    private formBuilderLogin: FormBuilder,
+    private formBuilderRegister: FormBuilder,
+    private router: Router,
+    private route: ActivatedRoute,
+    private authService: AuthService,
+    private alertService: AlertService
+  ) {
+    this.selectedTabIndex = 1;
+    if (this.authService.currentUserValue) {
+      this.router.navigate(["/"]);
+    }
+  }
+
+  ngOnInit() {
+    this.loginForm = this.formBuilderLogin.group({
+      email: ["", Validators.required],
+      password: ["", Validators.required]
+    });
+
+    // this.returnUrl = this.route.snapshot.queryParams["returnUrl"] || "/";
+
+    this.registerForm = this.formBuilderRegister.group({
```

Comment:
I think, better to move on register and login business login into a 2 separate components to manage them easily